data/mods/ssb/scripts.ts

```diff
@@ -100,324 +212,650 @@ export const Scripts: ModdedBattleScriptsData = {
+				}
+				if (moveData.slotCondition) {
+					hitResult = target.side.addSlotCondition(target, moveData.slotCondition, pokemon, move);
+					didSomething = this.combineResults(didSomething, hitResult);
+				}
+				if (moveData.weather) {
+					hitResult = this.field.setWeather(moveData.weather, pokemon, move);
+					didSomething = this.combineResults(didSomething, hitResult);
+				}
+				if (moveData.terrain) {
+					hitResult = this.field.setTerrain(moveData.terrain, pokemon, move);
+					didSomething = this.combineResults(didSomething, hitResult);
+				}
+				if (moveData.pseudoWeather) {
+					hitResult = this.field.addPseudoWeather(moveData.pseudoWeather, pokemon, move);
+					didSomething = this.combineResults(didSomething, hitResult);
+				}
+				if (moveData.forceSwitch && !this.getAllActive().some(x => x.hasAbility('skilldrain'))) {
+					hitResult = !!this.canSwitch(target.side);
+					didSomething = this.combineResults(didSomething, hitResult);
+				}
+				// Hit events
+				//   These are like the TryHit events, except we don't need a FieldHit event.
+				//   Scroll up for the TryHit event documentation, and just ignore the "Try" part. ;)
+				if (move.target === 'all' && !isSelf) {
+					if (moveData.onHitField) {
+						hitResult = this.singleEvent('HitField', moveData, {}, target, pokemon, move);
+						didSomething = this.combineResults(didSomething, hitResult);
+					}
+				} else if ((move.target === 'foeSide' || move.target === 'allySide') && !isSelf) {
+					if (moveData.onHitSide) {
+						hitResult = this.singleEvent('HitSide', moveData, {}, target.side, pokemon, move);
+						didSomething = this.combineResults(didSomething, hitResult);
+					}
+				} else {
+					if (moveData.onHit) {
+						hitResult = this.singleEvent('Hit', moveData, {}, target, pokemon, move);
+						didSomething = this.combineResults(didSomething, hitResult);
+					}
+					if (!isSelf && !isSecondary) {
+						this.runEvent('Hit', target, pokemon, move);
+					}
+				}
 			}
-			if (this.battle.field.getPseudoWeather('distortionworld')) {
-				speed = 0; // Anything times 0 is still 0
+			if (moveData.selfSwitch && !this.getAllActive().some(x => x.hasAbility('skilldrain'))) {
+				if (this.canSwitch(pokemon.side)) {
+					didSomething = true;
+				} else {
+					didSomething = this.combineResults(didSomething, false);
+				}
 			}
-			return this.battle.trunc(speed, 13);
-		},
-		isGrounded(negateImmunity = false) {
+			// Move didn't fail because it didn't try to do anything
+			if (didSomething === undefined) didSomething = true;
+			damage[i] = this.combineResults(damage[i], didSomething === null ? false : didSomething);
+			didAnything = this.combineResults(didAnything, didSomething);
+		}
+
+
+		if (!didAnything && didAnything !== 0 && !moveData.self && !moveData.selfdestruct) {
+			if (!isSelf && !isSecondary) {
+				if (didAnything === false) {
+					this.add('-fail', pokemon);
+					this.attrLastMove('[still]');
+				}
+			}
```

Comment:
```suggestion
			if (!isSelf && !isSecondary && didAnything === false) {
				this.add('-fail', pokemon);
				this.attrLastMove('[still]');
			}
```
There are a lot of unnecessarily nested `if` loops here :|